rootex/framework/entity.cpp

```diff
@@ -66,13 +89,32 @@ bool Entity::setupComponents()
 bool Entity::setupEntities()
 {
 	bool status = true;
+
+	if (m_Script)
+	{
+		bool result = m_Script->setup();
+		if (!result)
+		{
+			m_Script.reset();
+		}
+	}
 	for (auto& component : m_Components)
 	{
 		status = status & component.second->setupEntities();
 	}
 	return status;
 }
 
+sol::table Entity::getScriptEnv()
+{
+	return (sol::table)m_Script->m_ScriptEnvironment;
```

Comment:
Does `Entity` need to access the script env directly like that? `m_Script->getEnv()` looks much cleaner and possible better considering how `Entity` and `Script` are related but they shouldn't be friends